drivers/sensor/veml7700/veml7700.c

```diff
@@ -535,30 +499,80 @@ static int veml7700_init(const struct device *dev)
 	return 0;
 }
 
+static int veml7700_init(const struct device *dev)
+{
+	/* If the device is behind a power domain, it will start in
+	 * PM_DEVICE_STATE_OFF.
+	 */
+	if (pm_device_on_power_domain(dev)) {
+		pm_device_init_off(dev);
+		LOG_INF("Init %s as PM_DEVICE_STATE_OFF", dev->name);
+		return 0;
+	}
+
+	return veml7700_enable(dev);
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int veml7700_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct veml7700_config *conf = dev->config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		if (conf->psm != VEML7700_PSM_DISABLED) {
```

Comment:
I am still confused about the purpose of `VEML7700_PSM_DISABLED`. If you don't want runtime PM, just don't turn it on for the device?

Reply:
The psm register controls sampling rate and resolution of the sensor. You are right, that has nothing to do in veml7700_pm_action - removed it.